Kernel/Arch/riscv64/PCI/Initializer.cpp

```diff
@@ -180,24 +182,61 @@ void initialize()
                     auto prefetchable = (pci_address_metadata >> OpenFirmwareAddress::prefetchable_offset) & OpenFirmwareAddress::prefetchable_mask;
                     if (prefetchable)
                         continue; // We currently only use non-prefetchable 32-bit regions, since 64-bit regions are always prefetchable - TODO: Use 32-bit prefetchable regions if only they are available
-                    if (pci_32bit_mmio_base.has_value() && pci_32bit_mmio_size >= mmio_size)
+                    if (pci_32bit_mmio_size >= mmio_size)
                         continue; // We currently only use the single largest region - TODO: Use all available regions if needed
                     pci_32bit_mmio_base = mmio_address;
                     pci_32bit_mmio_size = mmio_size;
                 } else {
-                    if (pci_64bit_mmio_base.has_value() && pci_64bit_mmio_size >= mmio_size)
+                    if (pci_64bit_mmio_size >= mmio_size)
                         continue; // We currently only use the single largest region - TODO: Use all available regions if needed
                     pci_64bit_mmio_base = mmio_address;
                     pci_64bit_mmio_size = mmio_size;
                 }
             }
         }
+
+        auto maybe_interrupt_map = node.get_property("interrupt-map"sv);
+        auto maybe_interrupt_map_mask = node.get_property("interrupt-map-mask"sv);
+        if (maybe_interrupt_map.has_value() && maybe_interrupt_map_mask.has_value()) {
+            auto mask_stream = maybe_interrupt_map_mask.value().as_stream();
+            u32 metadata_mask = MUST(mask_stream.read_value<BigEndian<u32>>());
+            MUST(mask_stream.discard(sizeof(u32) * 2));
+            u32 pin_mask = MUST(mask_stream.read_value<BigEndian<u32>>());
+            interrupt_mask = ((metadata_mask >> 8) << 8) | pin_mask;
```

Comment:
Combining the b.d.f address and interrupt pin into one 32 bit value seems a bit weird and imo makes this code a bit harder to follow.

Reply:
I mean it is by default a combined 96 bit value, I just compressed it to fit into a word size 🤷. Converting it into a struct could also work, but would require implementing custom hash and equality functions etc, not sure all of that is worth getting rid a couple of shifts.

Reply:
Keeping this a 32 bit value does indeed make lookup easier.
Maybe add a comment to the PCIConfiguration struct saying that the keys of masked_interrupt_mapping contain the bus, device and function at the same place as Open Firmware PCI addresses and the register field is repurposed as the interrupt pin?

Reply:
sure, added a comment.